src/hooks/useAccommodationsAndOffers/helpers.ts

```diff
@@ -46,48 +46,47 @@ export const getActiveAccommodations = (
  * Get the lowest and highest price, total or per night/room from an array of offers
  *
  * @param offers offers array
- * @param perNightPerRoom if "true" get price per night/room, "false" to get total price
+ * @param perRoom if "true" get price per room, "false" to price for all requested rooms
+ * @param perNight if "true" get price per night, "false" to price for all requested nights
  * @param getPreferredCurrency return price ranges of the preferred currency where available
- * @param numberOfDays optional - required when perNightPerRoom is true
- * @param numberOfRooms optional - required when perNightPerRoom is true
+ * @param numberOfDays -- number of days
+ * @param numberOfRooms - number of rooms
  * @returns a PriceRange or undefined
  */
 export const getOffersPriceRange = (
   offers: OfferRecord[],
-  perNightPerRoom = true,
+  perRoom = true,
+  perNight = true,
   getPreferredCurrency = false,
-  numberOfDays?: number,
-  numberOfRooms?: number
+  numberOfDays: number,
+  numberOfRooms: number
 ): PriceRange | undefined => {
   let priceRange: PriceRange | undefined = undefined;
 
   offers
     .map((offer) => {
-      const price = getPreferredCurrency
+      //offer that we get from proxies is "total price per room"
+      const totalPricePerRoom = getPreferredCurrency
         ? offer.preferredCurrencyPrice?.public
         : offer.price.public;
       const currency = getPreferredCurrency
         ? offer.preferredCurrencyPrice?.currency
         : offer.price.currency;
 
-      if (!price || !currency) return;
-
-      if (perNightPerRoom) {
-        if (!numberOfDays || !numberOfRooms) {
-          throw new Error(
-            'Error: To get a nightly price range "numberOfDays" and "numberOfRooms" must be provided'
-          );
-        }
-        return {
-          price: Number(price) / (numberOfDays * numberOfRooms),
-          currency
-        };
-      } else {
-        return {
-          price: Number(price),
-          currency
-        };
+      if (!totalPricePerRoom || !currency) return;
+      let divider = 1;
+      let multiplier = 1;
+      //depending on selection (price per night or per rooms) we need to either divide or multiply
+      if (!perRoom) {
+        multiplier *= numberOfRooms;
+      }
+      if (perNight) {
+        divider *= numberOfDays;
       }
+      return {
+        price: (Number(totalPricePerRoom) / divider) * multiplier,
+        currency
+      };
```

Comment:
This is an awesome idea! The function handles 4 different evaluations depending on input. This is fine as it is.

Looking at it more closely now, as a suggestion, could we go one step further and just return all 4 evaluations in an object everytime, with the fields appropriately named `perRoomAllNights`, `perRoomPerNight`, `perNightAllRooms`, `AllRoomsAllNights`. 

Then it is up to the user to determine what they need. We will no longer need `perRoom` and `perNight` parameters. That way we probably won't mix up the parameters when we need a particular combination. and it is clear at first glance, what value we are using from the return of this function.